java/core/src/main/java/com/lancedb/lance/Fragment.java

```diff
@@ -101,6 +101,20 @@ public LanceScanner newScan(ScanOptions options) {
         dataset.allocator());
   }
 
+  /**
+   * Delete rows by row addresses. All the rows belong to the Fragment.
+   *
+   * @param rowAddrs The row addresses to delete.
+   * @return The fragment metadata after deletion. If all rows are deleted, return Null. Otherwise,
+   *     returns a new fragment with the updated deletion vector.
+   */
+  public FragmentMetadata deleteByAddrs(List<Long> rowAddrs) {
```

Comment:
I am starting to wonder if this binding is really needed, because there are many different ways to expose this, like why not convert addresses to indexes and then do JNI, or just only do JNI for writing the deletion vectors, or don't do anything in JNI. Do you have the corresponding PR for Spark Update/Merge? It probably makes more sense to review those together. What do you think?

Reply:
```
@Override
    public void delete(InternalRow metadata, InternalRow id) throws IOException {
      int fragmentId = metadata.getInt(0);
      deletedAddrs.compute(
          fragmentId,
          (k, v) -> {
            if (v == null) {
              v = new ArrayList<>();
            }
            v.add(id.getLong(0));
            return v;
          });
    }
```

is it possible for us to directly make `deletedAddrs` storing fragment ID -> list of row positions in the fragment ID? Becuase here you can already derive the row index from the address, that should be able to save the step in the commit to do another cycle of deriving the row index at JNI layer.